### PR TITLE
OCPBUGS-34041: Run metal3 containers as specific users and groups

### PR DESCRIPTION
**Pull request by MahnoorAsghar**
This is a cherry-pick of #417

### CODE REVIEW

**Comment by MahnoorAsghar** [comment]
Just realized that I can't simply rebase this PR off the master branch because this is a cherry pick PR for the release-4.16 branch. (Was rebasing in efforts to resolve the failing e2e-metal-ipi-upgrade-ovn-ipv6 branch, will revert the rebase after checking job results)

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.21-openshift-4.16
+  tag: rhel-9-release-golang-1.21-openshift-4.17
```
Comment (elfosardo) [inline .ci-operator.yaml:4]: let's not bump the images versions

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # When bumping the Go version, don't forget to update the configuration of the
 # CI jobs in openshift/release.
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.17 AS builder
```
Comment (elfosardo) [inline Dockerfile:3]: ditto and below too

**Review by elfosardo — CHANGES_REQUESTED** [review]

**Comment by MahnoorAsghar** [comment]
The e2e-metal-ipi-upgrade-ovn-ipv6 job is perma-failing for PRs targeting the release branches. I have opened a story in Jira to address this issue. Since this is an optional test and this bug is a release blocker, I would override this job on this PR.

**Pull request merged**